runtime/common/src/lib.rs

```diff
@@ -483,3 +483,78 @@ pub mod xcm_transactor {
 		}
 	}
 }
+
+pub mod foreign_investments {
+	use cfg_primitives::Balance;
+	use cfg_traits::SimpleCurrencyConversion;
+	use cfg_types::{
+		fixed_point::{FixedPointNumberExtension, Rate},
+		tokens::CurrencyId,
+	};
+	use frame_support::pallet_prelude::PhantomData;
+	use sp_runtime::{traits::Get, DispatchError};
+
+	/// Simple stable coin amount converter from one stable to another.
+	///
+	/// For now, converts any `ForeignAsset` into another with the configured
+	/// rate.
+	///
+	/// NOTE: Should be deprecated ASAP!
+	// TODO(@review): Can we determine whether a ForeignAsset is a stable coin at
+	// this point of time?
+	pub struct SimpleStableCurrencyConverter<RateForeignToPool>(PhantomData<RateForeignToPool>);
+
+	impl<RateForeignToPool> SimpleCurrencyConversion
+		for SimpleStableCurrencyConverter<RateForeignToPool>
+	where
+		RateForeignToPool: Get<Rate>,
+	{
+		type Balance = Balance;
+		type Currency = CurrencyId;
+		type Error = DispatchError;
+
+		fn foreign_to_pool(
+			currency_foreign: Self::Currency,
+			amount_foreign: Self::Balance,
+			currency_pool: Self::Currency,
+		) -> Result<Self::Balance, Self::Error> {
+			match (currency_foreign, currency_pool) {
+				(out, inc) if out == inc => Ok(amount_foreign),
+				// TODO(future): Conversion must be limited to asset ids which reflect stable coins
+				(CurrencyId::ForeignAsset(_out_id), CurrencyId::ForeignAsset(_in_id)) => {
+					// NOTE: Rounding down to favor system side
+					RateForeignToPool::get()
+						.checked_mul_int_floor(amount_foreign)
+						.ok_or(DispatchError::Arithmetic(
+							sp_arithmetic::ArithmeticError::Overflow,
+						))
+				}
+				_ => Err(DispatchError::Token(sp_runtime::TokenError::Unsupported)),
+			}
+		}
```

Comment:
A considerations about this (that I've already suffered with the oracle price conversions). I think at this point there are a couple of different conversions:
- One to convert the exchange rate of one stable to another (that should be always 1.0 because they're stable coins, and we are not trying to deal with the question "what happens if a stable coin drops?") 
- Another one to convert the decimal precisions used to represent each stable coin.

What of these values are handle by `RateForeignToPool` rate? First? Second? or Both? I would expect this rate to only handle the second point. Is that true?

Reply:
Right now, the `RateForeignToPool` represents an estimated exchange rate. It is true that in theory this should be `1.0` and thus not needed. But in practice, it might rather be `0.9...`.

What definitely is not needed in the `SimpleCurrencyConversion` are the current two separate conversions `foreign_to_pool` and `pool_to_foreign`, we only need `stable_to_stable` for now. Fixed in [d1d50be](https://github.com/centrifuge/centrifuge-chain/pull/1473/commits/d1d50bed5fc2ff240ea87021322887b45760c2ef)

Reply:
What would happen with the different decimal precision?

Suppose you want to convert the value `4.5` of `CurrencyIn` (measured with 3 decimals, so it's represented as `4_500`) into `CurrencyOut` (measured with 6 decimals). The resulting `Balance` should have the value `4_500_000`, supposing an exchange rate of `1.0`.

That "decimal" conversion is contained in the rate it is multiplied by?

Reply:
I think what you need is to use `AssetRegistry` to get the precision of both currencies and then call to [this method](https://github.com/centrifuge/centrifuge-chain/pull/1494/files#diff-833d286c0e9655fa1b12da1bceabc7ff8988d1c74b7149be553320af61b78b54R86)